sklearn/utils/_array_out.py

```diff
@@ -0,0 +1,16 @@
+import numpy as np
+
+
+def _get_feature_names(X):
+    """Get feature names from X.
+
+    Supports:
+       - pandas DataFrame
+       - xarray DataArray
```

Comment:
elaborate on the requirements for the xarray coords to be considered as feature names maybe